test/cron-job.test.ts

```diff
@@ -0,0 +1,296 @@
+import { Testing, ApiObject, Duration, CronOptions } from 'cdk8s';
+import { Node } from 'constructs';
+import * as kplus from '../src';
+
+test('defaultChild', () => {
+
+  const chart = Testing.chart();
+
+  const jobProperties = {};
+
+  const defaultChild = Node.of(new kplus.CronJob(chart, 'CronJob', {
+    jobProperties: jobProperties,
+  })).defaultChild as ApiObject;
+
+  expect(defaultChild.kind).toEqual('CronJob');
+});
+
+test('allow setting jobProperties / default configuration', () => {
+
+  const chart = Testing.chart();
+
+  new kplus.CronJob(chart, 'CronJob', {
+    jobProperties: {
+      activeDeadline: Duration.seconds(20),
+      backoffLimit: 4,
+      ttlAfterFinished: Duration.seconds(1),
+    },
+    containers: [{ image: 'image' }],
+  });
+
+  // assert the k8s spec has it.
+  const manifest = Testing.synth(chart);
+  expect(manifest).toMatchSnapshot();
+  const spec = manifest[0].spec;
+
+  console.log(`Spec: ${JSON.stringify(manifest, null, 2)}`);
+
+  expect(spec.schedule).toEqual('* * * * *');
+  expect(spec.concurrencyPolicy).toEqual('Forbid');
+  expect(spec.timeZone).toEqual(undefined);
+  expect(spec.startingDeadlineSeconds).toEqual(undefined);
+  expect(spec.suspend).toEqual(undefined);
+  expect(spec.successfulJobsHistoryLimit).toEqual(undefined);
+  expect(spec.failedJobsHistoryLimit).toEqual(undefined);
+
+  expect(spec.jobTemplate.spec.activeDeadlineSeconds).toEqual(20);
+  expect(spec.jobTemplate.spec.backoffLimit).toEqual(4);
+  expect(spec.jobTemplate.spec.ttlSecondsAfterFinished).toEqual(1);
+  expect(spec.jobTemplate.spec.template.spec.containers[0].image).toEqual('image');
+});
+
+test('allow setting schedule using cron options', () => {
```

Comment:
No need for a different test case for each property. Just have one test that doesn't set any custom value, and one test that sets all available custom values. Then snapshot test it.

Reply:
Great, will remove these.